designs/spot-consolidation.md

```diff
@@ -0,0 +1,105 @@
+# Spot Consolidation
+
+## Problem Statement
```

Comment:
Is it worth touching on the fact that most cloud providers don't surface capacity pool information for spot (for obvious reasons) which presents unique challenges that account for why we have to go with many of the solutions and their heuristics below

Reply:
Added a paragraph stating the reason for the requirement of heuristics in various solutions that we have suggested.